### ISSUE TITLE
clickhouse-local stops working on 21.4.6.55

### ISSUE THREAD

**Issue by q0987** [ClickHouse/ClickHouse#25897]
Hello all,

I cannot run clickhouse-local anymore on version 21.4.6.55 but I can run it without any issue on version 18.6.0.

Question 1> What is the issue here and How I can fix it?

Question 2> Why the clickhouse-local requires to write into config.xml?

Thank you

```
$ echo -e "1,2\n3,4" | clickhouse-local --structure "a Int64, b Int64" --input-format "CSV" --query "SELECT * FROM table"
Processing configuration file 'config.xml'.
Include not found: clickhouse_remote_servers
Include not found: clickhouse_compression
Couldn't save preprocessed config to /var/lib/clickhouse/preprocessed_configs/config.xml: Access to file denied: /var/lib/clickhouse/preprocessed_configs/config.xml
Logging trace to /var/log/clickhouse-server/clickhouse-server.log
Poco::Exception. Code: 1000, e.code() = 13, e.displayText() = Access to file denied: /var/log/clickhouse-server/clickhouse-server.log, Stack trace (when copying this message, always include the lines below):

0. Poco::FileImpl::handleLastErrorImpl(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0x12683f3c in /usr/bin/clickhouse
1. Poco::FileStreamBuf::open(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) @ 0x126954e2 in /usr/bin/clickhouse
2. Poco::FileOutputStream::FileOutputStream(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, unsigned int) @ 0x1269652d in /usr/bin/clickhouse
3. Poco::LogFileImpl::LogFileImpl(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0x1269f4c2 in /usr/bin/clickhouse
4. Poco::FileChannel::unsafeOpen() @ 0x126892ee in /usr/bin/clickhouse
5. Poco::FileChannel::open() @ 0x126891e1 in /usr/bin/clickhouse
6. Loggers::buildLoggers(Poco::Util::AbstractConfiguration&, Poco::Logger&, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&) @ 0x89c6483 in /usr/bin/clickhouse
7. DB::LocalServer::initialize(Poco::Util::Application&) @ 0x89036a5 in /usr/bin/clickhouse
8. Poco::Util::Application::run() @ 0x125cf046 in /usr/bin/clickhouse
9. mainEntryClickHouseLocal(int, char**) @ 0x890e50b in /usr/bin/clickhouse
10. main @ 0x87f1dce in /usr/bin/clickhouse
11. __libc_start_main @ 0x22505 in /usr/lib64/libc-2.17.so
12. _start @ 0x87bd06e in /usr/bin/clickhouse
 (version 21.4.6.55 (official build))

$ clickhouse-local --version
ClickHouse client version 21.4.6.55 (official build).

-rw-r--r-- 1 clickhouse clickhouse 16235 May 10 08:24 /var/lib/clickhouse/preprocessed_configs/config.xml
```

=============================
```
$ echo -e "1,2\n3,4" | clickhouse-local --structure "a Int64, b Int64" --input-format "CSV" --query "SELECT * FROM table"
1       2
3       4
$ clickhouse-local --version
ClickHouse client version 18.6.0.

-rw-r--r-- 1 root root 16142 Apr 15  2019 /var/lib/clickhouse/preprocessed_configs/config.xml

```

**Comment by alexey-milovidov** [comment]
TLDR: run `clickhouse-local` from another directory, without `config.xml`.

You are running `clickhouse-local`, it looks for a config in current directory. You are running it inside a directory where the server's config is located. But `clickhouse-server` config does not make sense for `clickhouse-local`.

In most cases `clickhouse-local` does not need any config at all.

**Comment by q0987** [comment]
This is not true. I didn't run clickhouse-local from a directory where it has config.xml.

**Comment by alexey-milovidov** [comment]
But it managed to find
`Processing configuration file 'config.xml'.`
anyhow.

Type `ls -l` in the current working directory.

**Comment by q0987** [comment]
@alexey-milovidov You are right. By accident, my home directory does have a copy of the config.xml. Thank you very much!
